starknet_py/net/account/account.py

```diff
@@ -79,26 +83,31 @@ def __init__(
         :param key_pair: Key pair that will be used to create a default `Signer`.
         :param chain: ChainId of the chain used to create the default signer.
         """
-        if chain is None and signer is None:
-            raise ValueError("One of chain or signer must be provided.")
-
         self._address = parse_address(address)
         self._client = client
 
-        if signer is not None and key_pair is not None:
-            raise ValueError("Arguments signer and key_pair are mutually exclusive.")
-
-        if signer is None:
-            if key_pair is None:
+        if signer is not None:
+            if chain is not None:
+                warnings.warn("Arguments signer and chain are mutually exclusive.")
```

Comment:
Let's reword the error message so it clearly indicates that this combination will raise an error in future versions.